### PR TITLE
Kibana 8.16.0 tests synchronization

### PR DESCRIPTION
**Pull request by Dzuming**
Kibana 8.16.0 tests synchronization

### CODE REVIEW

File: e2e-tests/cypress/e2e/Dev-tools.cy.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('Dev tools', () => {
 
     cy.log('should verify GET /_index_template successful with 403 status');
     DevTools.sendRequest('GET /_index_template/');
-    DevTools.verifyIf403Status();
+    DevTools.verifyIf200Status();
```
Comment (coutoPL) [inline e2e-tests/cypress/e2e/Dev-tools.cy.ts:19]: wow, this looks strange. Why do we have 200 right now? Maybe, ES not pass sth it should not pass?

File: e2e-tests/cypress/support/page-objects/Spaces.ts [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
     cy.get('[data-test-subj=manageSpaces]').click({ force: true });
-    cy.get(`[data-test-subj="${spaceName} space-deleteSpace"]`).click({ force: true });
+    if (semver.gte(getKibanaVersion(), '8.16.0')) {
+      cy.get(`[id="${spaceNameLowerCaseAndDash}-actions"]`).click();
+      cy.get(`[data-test-subj="${spaceNameLowerCaseAndDash}-deleteSpace"]`).click();
+    } else {
+      cy.get(`[data-test-subj="${spaceName}-deleteSpace"]`).click();
+    }
+
+    cy.intercept(`s/${spaceNameLowerCaseAndDash}/api/spaces/space/${spaceNameLowerCaseAndDash}`).as('deleteSpace');
```
Comment (coutoPL) [inline e2e-tests/cypress/support/page-objects/Spaces.ts:18]: not only for 8.16.0?

File: e2e-tests/cypress/e2e/Dev-tools.cy.ts (hunk 1 above)
Comment (Dzuming) [reply e2e-tests/cypress/e2e/Dev-tools.cy.ts:19]: No, it was a problem with the test assertion it remembered the previous call result, that's why it was 403 (but it should be 200). Now we are waiting  for the request to start here https://github.com/beshu-tech/readonlyrest-e2e-tests/pull/15/files#diff-78852126a0872b210632d24b016a1bddb6cdc70a6296b35ae7ceac9150f33742R38

File: e2e-tests/cypress/support/page-objects/Spaces.ts (hunk 2 above)
Comment (Dzuming) [reply e2e-tests/cypress/support/page-objects/Spaces.ts:18]: No, it's valid for all versions, so after deleting, we will wait for ES's 204 status to detect if the delete request was successful.

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**